examples/EquivariantFilter.h

```diff
@@ -0,0 +1,351 @@
+                  std::true_type{});
+
+  template <typename>
+  static std::false_type test(...);
+
+ public:
+  static constexpr bool value = decltype(test<Geometry>(0))::value;
+};
+
+template <typename Geometry>
+class has_inputMatrix {
+ private:
+  template <typename G>
+  static auto test(int)
+      -> decltype(G::inputMatrix(std::declval<typename G::GType>()),
+                  std::true_type{});
+
+  template <typename>
+  static std::false_type test(...);
+
+ public:
+  static constexpr bool value = decltype(test<Geometry>(0))::value;
+};
+
+template <typename Geometry>
+class has_processNoise {
+ private:
+  template <typename G>
+  static auto test(int)
+      -> decltype(G::processNoise(std::declval<const typename G::Input&>()),
+                  std::true_type{});
+
+  template <typename>
+  static std::false_type test(...);
+
+ public:
+  static constexpr bool value = decltype(test<Geometry>(0))::value;
+};
+
+template <typename Geometry>
+class has_inputMatrixBt {
+ private:
+  template <typename G>
+  static auto test(int)
+      -> decltype(G::inputMatrixBt(std::declval<typename G::GType>()),
+                  std::true_type{});
+
+  template <typename>
+  static std::false_type test(...);
+
+ public:
+  static constexpr bool value = decltype(test<Geometry>(0))::value;
+};
+
+template <typename Geometry>
+class has_measurementMatrixC {
+ private:
+  template <typename G>
+  static auto test(int)
+      -> decltype(G::measurementMatrixC(std::declval<const Unit3&>(),
+                                        std::declval<int>()),
+                  std::true_type{});
+
+  template <typename>
+  static std::false_type test(...);
+
+ public:
+  static constexpr bool value = decltype(test<Geometry>(0))::value;
+};
+
+template <typename Geometry>
+class has_outputMatrixDt {
+ private:
+  template <typename G>
+  static auto test(int)
+      -> decltype(G::outputMatrixDt(std::declval<int>(),
+                                    std::declval<typename G::GType>()),
+                  std::true_type{});
+
+  template <typename>
+  static std::false_type test(...);
+
+ public:
+  static constexpr bool value = decltype(test<Geometry>(0))::value;
+};
+
+//========================================================================
+// Equivariant Filter (EqF)
+//========================================================================
+
+/// Equivariant Filter (EqF) implementation
+template <typename G, typename M, typename Geometry>
+class EqF {
+ private:
+  G X_hat;                // Filter state
+  Matrix Sigma;           // Error covariance
+  M xi_0;                 // Origin state
+  Matrix Dphi0;           // Differential of phi at origin
+  Matrix InnovationLift;  // Innovation lift matrix
+
+ public:
+  /**
+   * Initialize EqF
+   * @param Sigma Initial covariance
+   * @param m Number of sensors
+   */
+  EqF(const G& X0, const M& x0, const Matrix& Sigma, int m);
```

Comment:
X0 and x0 are related by x0 = \phi(X0, ref), and if it's an action, we can always get X0 from  this. So I think: we need to give 2 M thingies: x0, and ref. X0 is derived.

Because the state is defined on the manifold, the user will want to specify the initial state on the manifold. And, more often than not, they will initialize the filter *at* the reference element.

EqF(const M& xi_circle, const M& xi_0, const Matrix& Sigma, int m);
EqF(const M& xi_circle, const Matrix& Sigma, int m);